homeassistant/components/jewish_calendar/calendar.py

```diff
@@ -0,0 +1,247 @@
+"""Jewish Calendar calendar platform."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import UTC, date, datetime, timedelta
+import logging
+
+from hdate import HDateInfo, Zmanim
+
+from homeassistant.components.calendar import (
+    CalendarEntity,
+    CalendarEntityDescription,
+    CalendarEvent,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import (
+    CONF_DAILY_EVENTS,
+    CONF_LEARNING_SCHEDULE,
+    CONF_YEARLY_EVENTS,
+    DEFAULT_CALENDAR_EVENTS,
+    DailyCalendarEventType,
+    LearningScheduleEventType,
+    YearlyCalendarEventType,
+)
+from .entity import JewishCalendarConfigEntry, JewishCalendarEntity
+
+_LOGGER = logging.getLogger(__name__)
+PARALLEL_UPDATES = 0
+
+
+@dataclass(frozen=True, kw_only=True)
+class JewishCalendarCalendarEntityDescription(CalendarEntityDescription):
+    """Jewish Calendar calendar entity description."""
+
+    create_event_fn: Callable[
+        [str, date, HDateInfo, Zmanim], list[CalendarEvent] | CalendarEvent | None
```

Comment:
Would it make sense to replace the str type with `DailyCalendarEventType | YearlyCalendarEventType`